### DISCUSSION TITLE
Disable DHCP in cockpit

### DISCUSSION THREAD

**Discussion by cooka90** [cockpit-project/cockpit#16266] [Q&A] [answered]
Hi all,

We are running Cockpit on Raspbian 10 Buster which works really well. However, we have a small problem (nothing major) when we give our raspberry pi a static IP, it still shows DHCP enabled in cockpit with two ip address assigned - one address is our static and the other is from dhcp. Are we able to disable dhcp in cockpit?

Thanks

Edit: had a look through the discussion thread, couldn't see anything similar to this problem. Apologies if there is!

**Comment by martinpitt** [comment, marked as answer]
You can enable/disable DHCP in the IPv4 and/or IPv6 settings of the interface (the "edit" link):

![image](https://user-images.githubusercontent.com/200109/130947895-3bc27834-93e6-4adc-bed6-d37da58ea77e.png)

Does that work for you?

**Reply by cooka90** [reply]
Hi Martin, thanks for your reply. Yes that does work, however I wondered if it would be possible to have it set to manual by default and not automatic. Again its not a massive burden, but if we ever reimaged the pi we would just need to remember to set this back to manual again.

**Reply by martinpitt** [reply]
This default isn't something that cockpit controls -- that's NetworkManager. Cockpit merely shows the current state.

When building pre-configured images, you shouldn't use Cockpit (as that's an interactive human tool), but something like preseed, cloud-init, Ansible, etc. Either call `nmcli` or ship a /etc/NetworkManager/conf.d/ snippet to set up that interface.

**Reply by martinpitt** [reply]
... or /etc/NetworkManager/system-connections/ file rather, which seems more appropriate for configuring a concrete ethernet interface.

**Reply by cooka90** [reply]
Many thanks. Will take a look!
